### PR TITLE
Demonic Sheep Addon HUD

### PR DESCRIPTION
**Pull request by ZenBre4ker**
Added a fully scalable Octagonal hud for the demonic Sheep. Tested with the newest Github Version.

### CODE REVIEW

File: lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		local a = Color1.a
+
+		r = r + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.r - r) / (maxSteps - 1)
+		g = g + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.g - g) / (maxSteps - 1)
+		b = b + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.b - b) / (maxSteps - 1)
+		a = a + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.a - a) / (maxSteps - 1)
+
+		return Color(r, g, b, a)
+	end
+end
```
Comment (TimGoll) [inline lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua:159]: please add an empty line here

**Review by TimGoll — CHANGES_REQUESTED** [review]

File: lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua (hunk 1 above)
Comment (ZenBre4ker) [reply lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua]: Added it, is there a good reason for it, besides the linter?

Comment (saibotk) [reply lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua]: It is common practice.
See https://stackoverflow.com/a/729795

Comment (ZenBre4ker) [reply lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua]: Thanks, now I get it ^^

**Pull request merged**